idaes/surrogate/tests/test_pysmo_surrogate.py

```diff
@@ -0,0 +1,2110 @@
+                                                    '[82932.15557448889, -341158.6596663887, 516416.7835787105, -341158.659633822, 82932.15555811858], '
+                                                    '[121970.94144067129, -130593.32429416949, -341158.6596220617, 589986.4680877628, -240227.09701875152], '
+                                                    '[-73364.60163552182, 121970.94144804058, 82932.15555219717, -240227.09701673465, 108729.13454474375]], '
+                        '"optimal_y_mu": [[-3.999999999902883], [-1.999999999902883], [9.711698112369049e-11], [2.000000000097117], [4.000000000097117]], '
+                        '"training_R2": 0.9999962956250228, "training_rmse": 0.005443803800474329}, '
+                '"map": {"x_data_columns": "list", "x_data": "numpy", "x_data_min": "numpy", "x_data_max": "numpy", '
+                        '"x_data_scaled": "numpy", "optimal_weights": "numpy", "optimal_p": "str", "optimal_mean": "numpy", '
+                        '"optimal_variance": "numpy", "regularization_parameter": "str", "optimal_covariance_matrix": "numpy", '
+                        '"covariance_matrix_inverse": "numpy", "optimal_y_mu": "numpy", "training_R2": "str", "training_rmse": "str"}}}, '
+    '"input_labels": ["x1", "x2"], '
+    '"output_labels": ["z1", "z2"], '
+    '"input_bounds": {"x1": [0, 5], "x2": [0, 10]}, '
+    '"surrogate_type": "kriging"}'
+    )
+
+
+
+class TestSurrogateTrainingResult:
+
+    @pytest.fixture
+    def pysmo_outputs(self):
+        data = {'x1': [1, 2, 3, 4, 5], 'x2': [5, 6, 7, 8, 9], 'z1': [10, 20, 30, 40, 50]}
+        data = pd.DataFrame(data)
+
+        init_pr = pr.PolynomialRegression(data, data, maximum_polynomial_order=1, overwrite=True, multinomials=True)
+        vars = init_pr.get_feature_vector()
+        init_pr.training()
+
+        init_rbf = rbf.RadialBasisFunctions(data, basis_function='linear', overwrite=True)
+        init_rbf.get_feature_vector()
+        init_rbf.training()
+
+        init_krg = krg.KrigingModel(data, numerical_gradients=True, overwrite=True)
+        init_krg.get_feature_vector()
+        init_krg.training()
+
+        return init_pr, init_rbf, init_krg, vars
+
+
+    @pytest.mark.unit
+    def test_init(self):
+        init_func = SurrogateTrainingResult()
+        assert init_func.metrics == {}
+        assert init_func._model == None
+        assert init_func.expression_str == ""
+
+    @pytest.mark.unit
+    def test_model(self, pysmo_outputs):
+
+        out1, out2, out3, vars = pysmo_outputs
+
+        init_func_poly = SurrogateTrainingResult()
+        init_func_poly.model = out1
+        assert init_func_poly.expression_str == str(out1.generate_expression([vars[i] for i in vars.keys()]))
+        assert init_func_poly._model is not None 
+        assert isinstance(init_func_poly._model, pr.PolynomialRegression)
+        assert init_func_poly._model == out1
+
+
+        init_func_rbf = SurrogateTrainingResult()
```

Comment:
As noted above, if one of the assets fails for the polynomial case above, these asserts will not be run so we have no way of knowing if there is also a failure here.

Reply:
Fixed.